analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -202,19 +203,40 @@ private void addOrUpdateActivity(UserAnonymizedEntityHolder userAnonymizedHolder
 	private void addOrUpdateDayTruncatedActivity(UserAnonymizedEntityHolder userAnonymizedHolder, ActivityPayload payload,
 			GoalDto matchingGoal)
 	{
-		ActivityDto lastRegisteredActivity = getLastRegisteredActivity(payload, matchingGoal);
+		Optional<ActivityDto> lastRegisteredActivity = getLastRegisteredActivity(payload, matchingGoal);
+		if (precedesLastRegisteredActivity(payload, lastRegisteredActivity))
```

Comment:
Suggestion: For readability, I would move this down under the most common case (`if (canCombineWithLastRegisteredActivity...`).

Reply:
I put it as first because it goes before the combinable activity. I don't have a strong opinion about it.

Reply:
OK, that is also a good reason to order it like that, I will leave it.